compiler/modules/front/codeGen.bal

```diff
@@ -359,11 +360,40 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialE
             env.assignments.push(...effect.assignments);
         }
     }                
+    check unusedLocalVariables(cx, env, initialEnv.bindings);
     int[] assignments = [];
     addAssignments(assignments, env.assignments, startRegister);
     return { block: curBlock, assignments };
 }
 
+function unusedLocalVariables(CodeGenContext cx, Environment env, Binding? bindingLimit) returns CodeGenError? {
+    Binding? binding = env.bindings;
+    while true {
+        if binding === bindingLimit {
+            break;
+        }
+        else {
+            // binding is non-nil
+            Binding tem = <Binding>binding;
+            if isUnused(tem) {
```

Comment:
I think you can just do `if tem.unnarrowed == () && !tem.used`. The unnarrowed binding must have been added before bindingLimit; you can just ignore the narrowed one.

Reply:
Fixed